### ISSUE TITLE
ignore_for_function should be supported

### ISSUE THREAD

**Issue by vanlooverenkoen** [dart-lang/linter#1610]
ignore_for_file is handy if you want to ignore a lint warning for a specific file.

I tought It would be usefull to have an ignore statement for functions as well.

`ignore_for_function: ...` is what I had in mind.

**Comment by srawlins** [comment]
Yeah other linters (e.g. scss-lint) have the concept of ignoring for a whole "scope" or "block." If we implemented this support, I'd suggest one identifier for this concept, like `ignore_for_scope`, that handles ignoring within a whole class, function, if-block, while-loop, etc.

**Comment by srawlins** [comment]
Although, regarding the priority or usefulness of this idea, what is your specific case,  @vanlooverenkoen ? What rule would you like ignored for a whole function, but not for all code in a file, and which would need to be non-ergonomically ignored on multiple lines within that function?

**Comment by vanlooverenkoen** [comment]
As I pass arguments to from my navigator to my flutter widget I want to use an object as an argument. All my routes are defined in a generate route method and there I am using (as WidgetParam) so I would like to ignore it for this method but not for the rest of my file or project

**Comment by jaredbaszler** [comment]
`ignore_for_file` should also be a file in the context menu.
